src/main/java/fr/openent/competences/helpers/NoteControllerHelper.java

```diff
@@ -0,0 +1,32 @@
+package fr.openent.competences.helpers;
+
+import fr.openent.competences.constants.Field;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+
+public class NoteControllerHelper {
+
+    protected static final Logger log = LoggerFactory.getLogger(DevoirControllerHelper.class);
+
+    public static JsonObject setResponseExportReleve(JsonObject responseAnnual, JsonObject responsePeriodic) {
```

Comment:
C'est assez subtil mais tu retournes un JsonObject (`responsePeriodic`)  et que tu set en même temps...

Reply:
Etant donné que tu boucles x fois annualStudents par periodicStudent, est-ce qu'on ne pourrait pas  changer l'algo pour faire en sorte que ça te retourne un objet propre sur les studentsPeriodic que tu aimerais 'réassigner' ?

@qpic-cgi @BastienBenard vous en pensez quoi pour l'adaptation avec les 2 for ?

Reply:
Pour le typage de retour, j'aurais mis un void. Premièrement car tu n'en as pas besoin vu que tu retourne l'instance passer en paramètre (responsePeriodic), deuxièmes pour bien faire comprendre que cette méthode va interagir/modifier les instance passer en paramètre (a noté en java doc)

Reply:
@JinDouang, ce que tu appelles un objet propre, c'est un objet modélisé ?